### PR TITLE
Enable translating Snackbar errors in GameView

### LINKED ISSUE #764: [Feature]: Enable translating Snackbar errors in GameView
### Feature Summary

We should revise the way the in-game error messages are sent from the server to support translating them clientside into the `BaseSnackbar` that's shown

### Detailed Description

This is a follow up to #584 and #758 to enable translating of error messages in the `GameView` by revising all the error messages sent to the client in the `api/controllers/game` directory so that instead of directly containing the error message to show to the client (in English), the server generated error's `message` attribute has the key to the localization dictionary which the client can then translate with `vue-i18n`'s `t()` function. See #747 for an example implementation in a different page.

### PR DESCRIPTION
**Pull request by Posoroko**
## #764

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #764

## Please check the following

- [ ] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [ ] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

## Please describe additional details for testing this change
In the api/controlers/game folder, I modified every error message that are being passed to the handleError method in the gameVIew file.
Other files in the api/controlers/game folder contain error messages, but I'm not sure if they are displayed in the snackbar.

### CODE REVIEW

File: api/controllers/game/seven/untargeted-one-off.js [hunk 1]
```diff
@@ -77,16 +77,16 @@ module.exports = function (req, res) {
             }
             default:
               return Promise.reject({
-                message: 'You cannot play that card as a ONE-OFF without a target',
+                message: 'game.snackbar.oneOffs.cantPLayWithoutATArget',
```
Comment (Haviles04) [inline api/controllers/game/seven/untargeted-one-off.js:80]: I think Target has two capital letters

File: api/controllers/game/targeted-one-off.js [hunk 2]
```diff
@@ -39,12 +39,12 @@ module.exports = function (req, res) {
                     break;
                   }
                   return Promise.reject({
-                    message: "Your opponent's queen prevents you from targeting their other cards",
+                    message: 'game.snackbar.onneOffs.targetWithQueen',
```
Comment (Haviles04) [inline api/controllers/game/targeted-one-off.js:42]: One offs has 2 ns

**Review by Haviles04 — COMMENTED** [review]
Nice work! Found a couple spelling errors, but otherwise looks good

**Review by itsalaidbacklife — COMMENTED** [review]
Really nice work. I like your organization of game.snackbar with the global section and then the errors specific to a particular type of move as their own category

File: api/controllers/game/seven/jack.js [hunk 3]
```diff
@@ -49,22 +49,12 @@ module.exports = function (req, res) {
...
-              }
-              return Promise.reject({
-                message: "Your opponent's queen prevents you from targeting their other cards",
-              });
-            default:
-              return Promise.reject({
-                message: 'You cannot play a targeted one-off when your opponent has more than one Queen',
+          if(queenCount >= 1) {
+            return Promise.reject({
+                message: 'game.snackbar.global.blockedByQueen',
```
Comment (Haviles04) [inline api/controllers/game/seven/jack.js:54]: I think we may still need the conditional for if there is only one

File: src/translations/en.json [hunk 4]
```diff
@@ -209,6 +209,50 @@
...
+        "mustTargetPointCard": "You can only scuttle a card your opponent has played for points"
+      },
+      "oneOffs": {
+        "scrapIsEmpty": "You can only play a 3 ONE-OFF if there are cards in the scrap pile",
+        "opponentHasNoCards": "You cannot play a 4 as a ONE-OFF while your opponent has no cards in hand",
+        "sevenWithEmptyDeck": "You can only play a 7 as a ONE-OFF if there are cards in the deck",
+        "sevenWithOneCardInDeck": "You cannot play a 7 as a ONE-OFF if it is the last card in the deck",
+        "cantPlayWithoutATarget": "You cannot play that card as a ONE-OFF without a target",
+        "blockedByQueen": "Your opponent's queen prevents you from targeting their other cards",
+        "blockedByMultipleQueens": "You cannot play a Targeted One-Off (Two, Nine) when your opponent has more than one Queen",
```
Comment (itsalaidbacklife) [inline src/translations/en.json:252]: Last nitpick:these are the same as the ones in the game.snackbar.global block, right? Let's remove these and use those instead

**Review by itsalaidbacklife — APPROVED** [review]
Nice work! Just one nitpick around the queen messages. The ones in global and oneOff are the same I think, so let's combine them

File: api/controllers/game/seven/targeted-one-off.js [hunk 5]
```diff
@@ -36,11 +36,11 @@ module.exports = function (req, res) {
                   break;
                 }
                 return Promise.reject({
-                  message: "Your opponent's queen prevents you from targeting their other cards",
+                  message: 'game.snackbar.oneOffs.blockedByQueen',
                 });
               default:
                 return Promise.reject({
-                  message: 'You cannot play a targeted one-off when your opponent has more than one Queen',
+                  message: 'game.snackbar.oneOffs.blockedByMultipleQueens',
```
Comment (itsalaidbacklife) [inline api/controllers/game/seven/targeted-one-off.js:43]: These look like the only two consumers of the redundant oneOff version of this text

**Review by itsalaidbacklife — COMMENTED** [review]
Nice work! Just one nitpick around the queen messages. The ones in global and oneOff are the same I think, so let's combine them

**Comment by Haviles04** [comment]
LGTM

**Review by itsalaidbacklife — APPROVED** [review]
Excellent work. We have almost translated the application!

**Pull request merged**